build-system/check-package-manager.js

```diff
@@ -237,7 +237,8 @@ function runGulpChecks() {
   }).trim();
   const wrongGulp =
     !defaultGulpPath.includes('yarn') &&
-    !defaultGulpPath.includes('amphtml/node_modules');
+    !firstInstall &&
+    !defaultGulpPath.includes(fs.realpathSync('node_modules'));
```

Comment:
The `gulp-cli` path is expected to resolve even before `yarn` is called for the first time. Is that not the case in the Lando workflow?

Reply:
Only according to the current [setup instructions](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server), which say to install Gulp globally:

> ![image](https://user-images.githubusercontent.com/134745/63195464-7e7eaf80-c027-11e9-9c78-9cd8a8fcfbbd.png)

What about the local `gulp` being installed by `yarn`? Specifically, this facilitates the alternative instructions:

> An alternative to installing `gulp-cli` is to invoke each Gulp command in this guide with `npx gulp` during local development. This will also use the version of `gulp` packaged with the amphtml repository.

Ideally we should use the local version of Gulp that is being installed already rather than a global version as this ensures the developer uses the expected version.

This change was introduced in 35a754c12b73da6d0aae446cbf4317c1821726c9 to fix prevent the warning:

> Found gulp in an unexpected location' when repo in non-amphtml dir

Reply:
> Only according to the current [setup instructions](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server), which say to install Gulp globally:

I think there might be a confusion here between `gulp` and `gulp-cli` here. It's undesirable to use a global install of `gulp` due to the version problem you mentioned. However, it is desirable to use a global install of `gulp-cli`, which uses the locally installed version of `gulp` for each repo.

> Ideally we should use the local version of Gulp that is being installed already rather than a global version as this ensures the developer uses the expected version.

This is exactly what the setup instructions recommend.

Reply:
> it is desirable to use a global install of `gulp-cli`, which uses the locally installed version of `gulp` for each repo.

I see. I wasn't aware of that.

Nevertheless, if I decided to not install a global `gulp-cli` and instead just use `npx gulp` then I was seeing this warning:

```
WARNING: Could not find gulp-cli.
⤷ To install it, run "yarn global add gulp-cli".
⤷ See https://medium.com/gulpjs/gulp-sips-command-line-interface-e53411d4467 for more information.
```

So my change here is intended to eliminate that.

Reply:
This warning is intentional, and is based on the currently recommended setup that uses `gulp-cli` to run all our [development / testing tasks](https://github.com/ampproject/amphtml/blob/master/contributing/TESTING.md#testing-commands). The reason it was put it in place was because developers who were initially using a global `gulp` were seeing mysterious build failures when they uninstalled the global `gulp`, but did not replace it with `gulp-cli` or `npx`.

I don't think we should eliminate this warning. I'm good with re-wording it to include a note on `npx gulp`. WDYT?

```
WARNING: Could not find gulp-cli.
⤷ To use the locally installed version of gulp, invoke your tasks with "npx gulp".
⤷ To install gulp-cli, run "yarn global add gulp-cli".
⤷ See https://medium.com/gulpjs/gulp-sips-command-line-interface-e53411d4467 for more information.```

Reply:
Love it. I'll make that change.

Reply:
@rsimha Oh, but I believe this part of the change should remain:

```diff
- !defaultGulpPath.includes('amphtml/node_modules');
+ !defaultGulpPath.includes(fs.realpathSync('node_modules'));
```

This accounts for when someone clones the repo to a directory other than `amphtml`. Right?

Reply:
And should it perhaps be a cyan notice instead of a yellow warning?

Reply:
Let me do some testing on this and get back to you. I agree with the overall goal of making these warnings less annoying when someone is using a genuine alternative workflow like `npx gulp`, while being useful to those who are trying to use our default recommend workflow. 

Let me send you a separate PR to change `build-system/check-package-manager.js`, and link to this discussion. Sound good? :)

Reply:
This was split off into #24060. PR merged.